saleor/discount/models.py

```diff
@@ -56,6 +56,7 @@ class Voucher(models.Model):
         decimal_places=settings.DEFAULT_DECIMAL_PLACES)
     # not mandatory fields, usage depends on type
     countries = CountryField(multiple=True, blank=True)
+    all_countries = models.BooleanField(default=False)
```

Comment:
Do we really need a new database field for that? Looking at the current backend implementation I see that we would only have to allow setting `""` value as a valid choice to `countries` input field in the API. This is how it works for the old dashboard.

Reply:
In other words, there should be one way to handle that logic. By adding a new database we're introducing a new way to solve a problem which was already somehow implemented in the existing code.

Reply:
I based this off `shippingZone` that has similar solution, but introduces an additional property. Also setting `""` sounds counter-intuitive that it enables all countries to be fair.

Reply:
So we have three ways - field `default` for shipping zones, `all_countries` for vouchers and `ANY_COUNTRY = ''` as a valid country choice (not available in API though). We want to have one common way to handle that behavior.

Reply:
I modified the `shop` query to additionally return `('', 'Any Country')`, but that probably means some bits need to be updated on PWA, as I believe the same query is used for the checkout addresses (probably an easy filter to remove countries without code).

Reply:
👍 Makes sense.